File: CivitasRuby/lib/controlador.rb
# encoding:utf-8
require_relative 'civitas_juego'
require_relative 'vista_textual'
module Civitas
  class Controlador
    def initialize(juego,vista)
      @juego = juego
      @vista = vista
    end

    def juega
      @vista.setCivitasJuego(@juego)
      while !@juego.final_del_juego
        @vista.actualizarVista
        @vista.pausa
        operacion = @juego.siguiente_paso
        if operacion != OperacionesJuego::PASAR_TURNO
          @vista.mostrarEventos
        end
        terminado = @juego.final_del_juego
        if !terminado
          case (operacion)
            when OperacionesJuego::COMPRAR
              respuesta = @vista.comprar
              if respuesta == Respuestas::SI
                @juego.comprar
                @juego.siguiente_paso_completado(operacion)
              end
            when OperacionesJuego::GESTIONAR
              @vista.gestionar
              op_inmobiliaria = OperacionInmobiliaria.new(@vista.getGestion, @vista.getPropiedad)
              case (@vista.getGestion)
                when GestionesInmobiliarias::VENDER
                  @juego.vender(op_inmobiliaria.get_num_propiedad)
                when GestionesInmobiliarias::HIPOTECAR
                  @juego.hipotecar(op_inmobiliaria.get_num_propiedad)
                when GestionesInmobiliarias::CANCELAR_HIPOTECA
                  @juego.cancelar_hipoteca(op_inmobiliaria.get_num_propiedad)
                when GestionesInmobiliarias::CONSTRUIR_CASA
                  @juego.construir_casa(op_inmobiliaria.get_num_propiedad)
                when GestionesInmobiliarias::CONSTRUIR_HOTEL
                  @juego.construir_hotel(op_inmobiliaria.get_num_propiedad)
                when GestionesInmobiliarias::TERMINAR
                  @juego.siguiente_paso_completado(operacion)
              end
            when OperacionesJuego::SALIR_CARCEL
              case(@vista.salir_carcel)
                when SalidasCarcel::PAGANDO
                  @juego.salir_carcel_pagando
                when SalidasCarcel::TIRANDO
                  @juego.salir_carcel_tirando
              end
              @juego.siguiente_paso_completado(operacion)
          end
        end
      end
      puts @juego.ranking.to_s
    end
  end
end

File: CivitasJava/src/GUI/Controlador.java
package GUI;

import Civitas.CivitasJuego;
import Civitas.OperacionesJuego;
import Civitas.GestionesInmobiliarias;

public class Controlador {

    private CivitasJuego juego;
    private CivitasView vista;

    Controlador(CivitasJuego juego, CivitasView vista) {
        this.juego = juego;
        this.vista = vista;
    }

    void juega() {
        OperacionesJuego operacion;
        Respuestas resp;
        GestionesInmobiliarias gest;
        int numProp;
        int indGest;
        
        //vista.setCivitasJuego(juego);
        vista.setCivitas(juego);
        
        while(!this.juego.finalDelJuego()){
            vista.actualizarVista();
            //vista.pausa();
            operacion = juego.siguientePaso();

            if (operacion != OperacionesJuego.PASAR_TURNO)
                vista.mostrarEventos();
            
            if (!juego.finalDelJuego()) {
                switch (operacion) {
                    case COMPRAR:
                        resp = vista.comprar();
                        if (resp == Respuestas.SI) {
                            juego.comprar();
                        }
                        juego.siguientePasoCompletado(operacion);
                        break;
                    /*case GESTIONAR:
                        vista.gestionar();
                        indGest = vista.getGestion();
                        numProp = vista.getPropiedad();
                        gest = GestionesInmobiliarias.values()[indGest];
                        OperacionInmobiliaria opInmobiliaria = new OperacionInmobiliaria(gest, numProp);
                        System.out.println(indGest);
                        switch (indGest) {
                            case 1:
                                juego.vender(opInmobiliaria.getNumPropiedad());
                                break;
                            case 2:
                                juego.hipotecar(opInmobiliaria.getNumPropiedad());
                                break;
                            case 3:
                                juego.cancelarHipoteca(opInmobiliaria.getNumPropiedad());
                                break;
                            case 4:
                                juego.construirCasa(opInmobiliaria.getNumPropiedad());
                                break;
                            case 5:
                                juego.construirHotel(opInmobiliaria.getNumPropiedad());
                                break;
                            case 0:
                                juego.siguientePasoCompletado(operacion);
                                break;
                        }
                        break;
                    case SALIR_CARCEL:
                        switch (vista.salirCarcel()) {
                            case PAGANDO:
                                juego.salirCarcelPagando();
                                break;
                            case TIRANDO:
                                juego.salirCarcelTirando();
                                break;
                        }
                        juego.siguientePasoCompletado(operacion);
                        break;*/
                    }
                }
            }
            juego.ranking().toString();
    }

}
